ts/screens/messages/MessagesHomeScreen.tsx

```diff
@@ -110,6 +121,27 @@ const MessagesHomeScreen = ({
 
   const isScreenReaderEnabled = useScreenReaderEnabled();
 
+  const isLollipopEnabled = useIOSelector(isLollipopEnabledSelector);
+  const showUnsupportedDeviceBanner =
+    isLollipopEnabled && publicKeyState === "error";
+  const unsupportedDevicesStatusComponent = showUnsupportedDeviceBanner && (
```

Comment:
Creating and using this component with its hook input in this way causes an extra re-refresh of the entire screen/component. We can create a standalone component that has all its needed dependencies inside and embed it directly on both `{isScreenReaderEnabled && unsupportedDevicesStatusComponent}` and `{!isScreenReaderEnabled && unsupportedDevicesStatusComponent}` (replacing `unsupportedDevicesStatusComponent`). 
In this way, when the hook changes its data, only the inner component will refresh, avoiding an extra rendering on this `MessagesHomeScreen` screen.
What are your thoughts on this?

Reply:
That's sounds good. Maybe we leave this for a brand new PR: